migtests/scripts/resumption.py

```diff
@@ -0,0 +1,260 @@
+                for ready in rlist:
+                    output = ready.readline()
+                    if not output:  # Exit if output is empty (end of process output)
+                        break
+                    full_output += output
+                if time.time() - start_time > 5:
+                    break
+
+            if full_output:
+                print(full_output.strip(), flush=True)
+
+            while True:
+                if process.poll() is not None:
+                    break  # Process has ended, exit loop
+
+                interrupt_interval_seconds = random.randint(
+                    resumption['min_interrupt_seconds'], 
+                    resumption['max_interrupt_seconds']
+                )
+                print(f"\nProcess will be interrupted in {interrupt_interval_seconds // 60}m {interrupt_interval_seconds % 60}s")
+                time.sleep(interrupt_interval_seconds)
+                print(f"\nInterrupting the import process (PID: {process.pid})")
+                process.send_signal(signal.SIGINT)
+
+                restart_wait_time_seconds = random.randint(
+                    resumption['min_restart_wait_seconds'], 
+                    resumption['max_restart_wait_seconds']
+                )
+                print(f"\nWaiting for {restart_wait_time_seconds // 60}m {restart_wait_time_seconds % 60}s before resuming...")
+                time.sleep(restart_wait_time_seconds)
+
+        except Exception as e:
+            print(f"Error occurred during import: {e}")
+            if process:
+                process.kill()
+            raise e
+        
+        finally:
+            if process and process.poll() is None:
+                print(f"Terminating process (PID: {process.pid})")
+                process.kill()
+                process.wait(timeout=30)
+
+    # Final import retry logic
+    print("\n--- Final attempt to complete the import ---")
+    
+    for _ in range(2): 
+        try:
+            print("\nVoyager command output:")
+
+            process = subprocess.Popen(
+                command, 
+                stdout=subprocess.PIPE, 
+                stderr=subprocess.PIPE, 
+                text=True
+            )
+
+            # Capture and print output
+            for line in iter(process.stdout.readline, ''):
+                print(line.strip())
+                sys.stdout.flush()
+
+            process.wait()
+
+            if process.returncode != 0:
+                raise subprocess.CalledProcessError(process.returncode, command)
+
+            break
+        except subprocess.CalledProcessError as e:
+            print("\nVoyager command error:")
+            for line in iter(process.stderr.readline, ''):
+                print(line.strip())
+                sys.stdout.flush()
+            time.sleep(30)
+    else:
+        print("Final import failed after 2 attempts.")
+        sys.exit(1)
+
+def validate_row_counts(row_count, export_dir):
```

Comment:
note for future: you can create a common python file that has such helper 
functions.